File: Etudes/LOTS/Projets/JAVALOTS/src/reseau/Cable.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package reseau;

import general.Aret;
import general.Graph;
import general.Sommet;

/**
 *
 * @author BIBOUH
 */
public class Cable extends Aret{

    public Cable() {
        super();
    }

    @Override
    public boolean compatibleGraph(Graph s) {
        if(s instanceof Reseau)
            return true;
        else 
            return false;
        
    }

    @Override
    public boolean compatibleSommet(Sommet s) {
        if(s instanceof Ordinateur)
            return true;
        else 
            return false;
    }
    
    
    
}

File: Etudes/LOTS/projetLot/CSharp/genericite/Informatique/Cable.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;

using General;

namespace Informatique
{

  public class Cable : Aret
  {
      public Cable(String name) {
		this.name=name;
	}

	override public bool compatibleGraphe(Graphe graphe) {
		return (graphe is Reseau);
	}

	override public bool compatibleSommet(Sommet sommet) {
		return (sommet is Ordinateur);
	}


  }

}  // end of namespace Informatique

